scripts/analyze_deps.py

```diff
@@ -242,18 +242,32 @@ def render_report(output_path, report_context):
             sys.exit(0)
 
     frozen = {}
+    overrides = {}
+    override_count = 0
     try:
         with io.open(frozen_filename, 'r', encoding='utf-8-sig') as frozen_file:
             for line in frozen_file:
-                req_name, spec = parse_req(line)
-                frozen[req_name] = [spec]
+                if line.startswith('#override'):
+                    _, lib_name, req_override = line.split(' ', 2)
+                    record_dep(overrides, req_override, lib_name)
+                    override_count += 1
+                elif not line.startswith('#'):
```

Comment:
I have no idea what this code really does so I'm probably barking up the wrong tree here, but what should happen if a line starts with `#` but not `#override`. Is that a valid case or should that case throw or log or something?

Reply:
Anything inside a requirements.txt that starts with a `#` is a comment, and should be ignored. Previously this script didn't adhere to that convention. This change allows comments to be written into the shared_requirements.txt (which is a file of my own devising, but which I want to keep compatible with the standard requirements.txt format).

That's also the reason that the override mechanism is implemented as a special type of comment. So if someone wanted to they can still run `pip install -r shared_requirements.txt`, or programmatically load it and interact with it as if it were a normal requirements.txt file.